pkg/pod/status.go

```diff
@@ -159,7 +165,28 @@ func MakeTaskRunStatus(ctx context.Context, logger *zap.SugaredLogger, tr v1.Tas
 	return *trs, merr.ErrorOrNil()
 }
 
-func setTaskRunStatusBasedOnStepStatus(ctx context.Context, logger *zap.SugaredLogger, stepStatuses []corev1.ContainerStatus, tr *v1.TaskRun, podPhase corev1.PodPhase, kubeclient kubernetes.Interface, ts *v1.TaskSpec) *multierror.Error {
+func setTaskRunStatusBasedOnSpireVerification(ctx context.Context, logger *zap.SugaredLogger, tr *v1.TaskRun, trs *v1.TaskRunStatus,
+	filteredResults []result.RunResult, spireAPI spire.ControllerAPIClient) {
+	if tr.IsSuccessful() && spireAPI != nil &&
+		((tr.Status.TaskSpec != nil && len(tr.Status.TaskSpec.Results) >= 1) || len(filteredResults) >= 1) {
```

Comment:
This condition is a bit hard to understand. Can we break it down a bit? Maybe something that can be reused on line 183.
i.e. 

```go
hasResults := tr.Status.TaskSpec != nil && len(tr.Status.TaskSpec.Results) >= 1
```